[Refactor] #124. 댓글 순서를 위한 매핑작업, [Fix] #126. 올바른 댓글 순서 보여주기

mooncw opened the pull request:
## 개요
프론트의 댓글 순서 문제를 java Map으로 해결하고자 함

### 상세 내용
- 백엔드의 PostEntitiy, PostDto, PostConverter 수정
- PostDto에서 매핑
- key값은 "0" or 본댓글의 id
- 프론트엔드의 post 수정

- 매핑 정보를 이용함
- 화면상 댓글을 지우면 해당하는 대댓글도 지워지나 DB에 해당 대댓글이 남아있는 문제가 있음
- 수정으로 인해 현재 post의 commentList 사용하지 않음

Resolves: #124 #126

## PR 유형
어떤 변경 사항이 있나요?

- [ ] 새로운 기능 추가
- [x] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제
- [ ] 배포 및 PR 관련

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다. [Commit message convention 참고](https://quickest-asterisk-75d.notion.site/2a977a0d71db4062bc705dc199ebff13?pvs=4).
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

mooncw committed:
[Refactor] #124. 댓글 순서를 위한 매핑작업

- 백엔드의 PostEntitiy, PostDto, PostConverter 수정
- key값은 "0" or 본댓글의 id
- 프론트엔드의 post 수정

ShineCorine reviewed (approved):
고생하셨습니다!

kkkapuq commented on backend/orury/src/main/java/com/kernel360/orury/domain/post/service/PostConverter.java:
```diff
@@ -32,6 +32,23 @@ public PostDto toDto(PostEntity postEntity) {
                 .toList();
 
+        // map으로 변환
+        Map<String, List<CommentDto>> commentMap = new HashMap<>();
```
여기 LinkedHashMap으로 해야될 것 같습니다.
오래된 댓글부터 위에서부터 순차적으로 뿌려줘야해서 순서 보장이 되어야해요

kkkapuq commented on frontend/lib/presentation/Board/post.dart:
```diff
@@ -27,13 +31,25 @@ class Post {
     var commentList = json['comment_list'] as List;
     List<Comment> commentObjects = commentList.map((comment) => Comment.fromJson(comment)).toList();
 
+    final Map<String, List<Comment>> commentMap2 = {};
```
Dart에선 넘어온 Map객체를 쓰기만 하면 되니 LinkedHashMap을 쓸 필요는 없어보입니다

mooncw committed:
[Fix] #126. 프론트에서 댓글 순서 올바르게 보여주기

- 매핑 정보를 이용함
- 화면상 댓글을 지우면 해당하는 대댓글도 지워지나 DB에 해당 대댓글이 남아있는 문제가 있음
- 수정으로 인해 현재 post의 commentList 사용하지 않음
